File: classic/Code/PalindromeNumber.java
public class Solution {
    public boolean isPalindrome(int x) {
        if(x<0)return false;
        if(x==0)return true;
        int n=0;
        int tmp=x;
        while(tmp>0){
            tmp/=10;
            n++;
        }
        for(int i=0;i<n;i++){
            int a=(int)(x/Math.pow(10,i))%10;
            int b=(int)(x/Math.pow(10,n-1-i))%10;
            if(a!=b)return false;
        }
        return true;
    }
}

File: classic/Code/PalindromeNumber.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0){return false;}
        int d=1;
        int tmp=x;
        while (tmp/10){
            tmp=tmp/10;
            d++;
        }
        for(int i=0;i<d/2;i++){
            int right=int(x/pow(10,i))%10;
            int left=int(x/pow(10,d-i-1))%10;
            if(right!=left){return false;}
        }
        return true;
    }
};
